src/Service.Tests/SqlTests/GraphQLMutationTests/MultipleMutationIntegrationTests.cs

```diff
@@ -0,0 +1,327 @@
+        /// no source of value for a non-nullable and non-default referencing column.
+        /// There are 3 sources of values for a referencing column:
+        /// 1. Explicit value assignment to the referencing column,
+        /// 2. Value derived from the parent referenced entity,
+        /// 3. Providing value for inserting record in the referenced target entity, in which case value for referencing column
+        /// is derived from the value of corresponding referenced column after insertion in the referenced entity.
+        /// When the user provides none of the above, we throw an exception.
+        /// </summary>
+        [TestMethod]
+        public async Task AbsenceOfSourceOfTruthForReferencingColumnForCreateMultipleMutations()
+        {
+            // For a relationship between Book (Referencing) - Publisher (Referenced) defined as books.publisher_id -> publisher.id,
+            // in the request input:
+            // 1. No explicit value is provided for referencing field publisher_id
+            // 2. Book is the top-level entity - so there is no parent entity.
+            // 3. No value is assigned for the referenced entity via relationship field ('publisher').
+            string createmultipleBooksMutationName = "createbooks";
+            string createMultipleBooksWithoutPublisher = @"mutation {
+                    createbooks(items: [{ title: ""My New Book"" }]) {
+                        items{
+                           id
+                           title
+                        }
+                    }
+                }";
+
+            JsonElement actual = await ExecuteGraphQLRequestAsync(
+                query: createMultipleBooksWithoutPublisher,
+                queryName: createmultipleBooksMutationName,
+                isAuthenticated: true,
+                variables: null,
+                clientRoleHeader: "authenticated");
+
+            // Validate that an appropriate exception is thrown because the value for the referencing field
+            // publisher_id cannot be derived from any of the two possible sources mentioned above.
+            SqlTestHelper.TestForErrorInGraphQLResponse(
+                    actual.ToString(),
+                    message: "Missing value for required column: publisher_id for entity: Book at level: 1.",
+                    statusCode: DataApiBuilderException.SubStatusCodes.BadRequest.ToString()
+                );
+        }
+
+        /// <summary>
+        /// Test to validate that for 'create one' mutations, we throw an appropriate exception when the user request provides
+        /// multiple sources of value for a referencing column. There are 3 sources of values for a referencing column:
+        /// -> Explicit value assignment to the referencing column,
+        /// -> Value derived from the parent referenced entity,
+        /// -> Providing value for inserting record in the referenced target entity, in which case value for referencing column
```

Comment:
do the three test cases in this testmethod exercise different validation functions? if so, could they be split up into separate test methods where each testmethod comment gives context as to what validation code (validation function(s)) is(are) exercised?

Reply:
I think the current setup makes sense in terms of keeping the tests with same concern together. I will add a comment to the test case which excercises a particular method.